### ISSUE TITLE
New USPS Zone Postage change needed for Canada delivery

### ISSUE THREAD

**Issue by Dirty-Butter** [cubecart/v6#617]
See https://forums.cubecart.com/topic/49895-resolved-usps-canada-parcel-rate-zone-change-causing-issue/

**Comment by abrookbanks** [comment]
![screen shot 2015-06-18 at 15 46 20](https://cloud.githubusercontent.com/assets/4520803/8233986/349d2edc-15d1-11e5-963f-8954bec5021e.png)

**Comment by abrookbanks** [comment]
I don't really see the logic in these changes. It's good that this works for you but I'm not sure this is going to work for everyone.

**Comment by Dirty-Butter** [comment]
Is it a USA sending and Canada delivery address issue, so it only applies if store is in the US? I'm not sure. Moving some of the code from one place to another was Bsmither's first attempt to fix - perhaps not necessary? As I understand it, It was specifying that Canada Zip be required that made it work. Bsmither needs to weigh in on this one. :) I'm just the messenger.

**Comment by abrookbanks** [comment]
Thanks Rosemary..

**Comment by Dirty-Butter** [comment]
I sent you a copy of the file again. I must have attached a backup, rather than the changed one.

**Comment by bhsmither** [comment]
1. I moved these two lines:
```
		$xml->startElement('Package');
		$xml->writeAttribute('ID',0);
```
from above the if($delivery...) statement into the if block and also into the else block. The reason is so that I could add a specific statement:
`$xml->writeElement('Revision',2);`
 ahead of those two. This order is dictated in the USPS manual.

N.B.: There is more than one way to get `$xml->writeElement('Revision',2);` in the correct order of elements.

The new USPS requirement is that shipments to Canada must now declare the United States' zip code of origin: `OriginZip`. And declaring `Revision 2` is required when using `OriginZip`.

The `DestinationPostalCode is still optional, and I included it as a comment for reference purposes only.
